src/reducers/address_by_ada_handle.rs

```diff
@@ -0,0 +1,114 @@
+use std::str::FromStr;
+
+use pallas::codec::utils::{Bytes, KeyValuePairs};
+use pallas::crypto::hash::Hash;
+use pallas::ledger::primitives::babbage::Value;
+use pallas::ledger::traverse::MultiEraBlock;
+use pallas::ledger::traverse::MultiEraOutput;
+use serde::Deserialize;
+
+use crate::{model, prelude::*};
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub key_prefix: Option<String>,
+    pub filter: Option<Vec<String>>,
+    pub policy_id_hex: String,
+}
+
+pub struct Reducer {
+    config: Config,
+    policy_id: Hash<28>,
+}
+
+type Asset = (Hash<28>, Bytes, u64);
+
+fn iter_policy_assets<'b>(
+    policy: &'b Hash<28>,
+    assets: &'b KeyValuePairs<Bytes, u64>,
+) -> impl Iterator<Item = Asset> + 'b {
+    assets
+        .iter()
+        .map(|(name, amount)| (policy.clone(), name.clone(), *amount))
+}
+
+// TODO: replace this with upstream pallas method
+fn collect_txo_assets(txo: &MultiEraOutput) -> Vec<Asset> {
```

Comment:
Why not immediately use function from @rvcas ?

Reply:
Pallas version not yet published, will refactor once we have it.